src/main/java/practice/StreamPractice.java

```diff
@@ -74,7 +102,11 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * We want to reuse our validation in future, so let's write our own impl of Predicate
      * parametrized with Candidate in CandidateValidator.
      */
-    public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+    public static List<String> validateCandidates(List<Candidate> candidates) {
+        return candidates.stream()
+                .filter(candidate -> new CandidateValidator().test(candidate))
```

Comment:
Можливо варто спробувати так?
```suggestion
                .filter(new CandidateValidator())
```

Reply:
done